docs/typescript.md

```diff
@@ -71,17 +71,31 @@ const App = () => (
 ```tsx
 import styled from 'react-emotion';
 
-const NotALink = styled('div')`
+const NotALink = styled<{}, 'div'>('div')`
   color: red;
 `
 
 const Link = NotALink.withComponent('a')
 
+type TextProps = {
+  className: string;
+  text: string;
+}
+const Text = ({ className, text }: TextProps) => <div className={className}>{text}</div>
+const StyledText = styled<{}, TextProps>(Text)`
+  fontSize: 20px;
+  color: white;
+`;
+
+const OtherLink = StyledText.withComponent('a');
+
 const App = () => <Link href="#">Click me</Link>
```

Comment:
Should `Link` be `OtherLink` in this example?

Reply:
We also have `Link`... Do I need to use all these components (`Link`, `OtherLink`, ...)?